test/box-luatest/gh_7583_txn_stat_test.lua

```diff
@@ -68,7 +68,10 @@ g.before_test('test_replication', function(cg)
     cg.replica = server:new({
         alias = 'replica',
         box_cfg = {
-            replication = server.build_listen_uri('default'),
+            replication = {
+                server.build_listen_uri('default'),
+                server.build_listen_uri('replica'),
```

Comment:
That looks like a strange restriction that we always have to include self into the replication. The simplest possible replication example (which I am 99% sure is even somewhere in our docs) now doesn't work:
```Lua
box.cfg{listen = 3313}
box.schema.user.grant('guest', 'super')
```
```Lua
box.cfg{listen = 3314, repliction={3313}}
```
Making this example an error is really weird. I would double-check with someone if this is really the intentional change.

Reply:
Agree. The point of checking registered nodes is in trying to connect to everyone. No need to connect to self, since this is a join and not replica set bootstrap.
I tried to remove all the excess changes in tests caused by this:
```diff
diff --git a/test/box-luatest/gh_6260_add_builtin_events_test.lua b/test/box-luatest/gh_6260_add_builtin_events_test.lua
index 89f691588..7dbef64eb 100644
--- a/test/box-luatest/gh_6260_add_builtin_events_test.lua
+++ b/test/box-luatest/gh_6260_add_builtin_events_test.lua
@@ -50,12 +50,13 @@ end
 
 g.before_test('test_box_status', function(cg)
     cg.replica_set = replica_set:new({})
-    cg.master = cg.replica_set:build_and_add_server({
+    cg.master = cg.replica_set:build_server({
         alias = 'master',
         box_cfg = {
             bootstrap_strategy = 'legacy',
         },
     })
+    cg.replica_set:add_server(cg.master)
     cg.replica_set:start()
 end)
 
@@ -150,9 +151,9 @@ g.before_test('test_box_election', function(cg)
         election_mode = 'off',
         replication_synchro_quorum = 2,
         replication_synchro_timeout = 1,
+        replication_sync_timeout = 300,
         replication_timeout = 0.25,
         election_timeout = 0.25,
-        bootstrap_strategy = 'legacy',
     }
 
     cg.instance_1 = cg.replica_set:build_server(
diff --git a/test/box-luatest/gh_7583_txn_stat_test.lua b/test/box-luatest/gh_7583_txn_stat_test.lua
index 67c7ac023..6b5ea66bc 100644
--- a/test/box-luatest/gh_7583_txn_stat_test.lua
+++ b/test/box-luatest/gh_7583_txn_stat_test.lua
@@ -68,10 +68,7 @@ g.before_test('test_replication', function(cg)
     cg.replica = server:new({
         alias = 'replica',
         box_cfg = {
-            replication = {
-                server.build_listen_uri('default'),
-                server.build_listen_uri('replica'),
-            },
+            replication = server.build_listen_uri('default'),
         },
     })
 end)
diff --git a/test/box-luatest/gh_7846_crash_on_replica_id_alter_test.lua b/test/box-luatest/gh_7846_crash_on_replica_id_alter_test.lua
index 6168652bf..83f6985fb 100644
--- a/test/box-luatest/gh_7846_crash_on_replica_id_alter_test.lua
+++ b/test/box-luatest/gh_7846_crash_on_replica_id_alter_test.lua
@@ -1,5 +1,6 @@
 local server = require('luatest.server')
 local t = require('luatest')
+local urilib = require('uri')
 
 local g = t.group('before_replace alter',
     t.helpers.matrix{ret_nil = {true, false}})
@@ -23,14 +24,10 @@ g.test_before_replace_alter_replica_id = function(cg)
         end
         box.space._cluster:before_replace(trigger)
     end, {ret_nil})
+    local uri = urilib.parse(server1.net_box_uri)
     local server2 = server:new{
         alias = 'server2',
-        box_cfg = {
-            replication = {
-                server.build_listen_uri('server1'),
-                server.build_listen_uri('server2'),
-            },
-        },
+        box_cfg = {replication = uri.unix}
     }
 
     if ret_nil == false then
diff --git a/test/box-luatest/gh_7988_auth_type_test.lua b/test/box-luatest/gh_7988_auth_type_test.lua
index d45d6ecea..19b69ae1d 100644
--- a/test/box-luatest/gh_7988_auth_type_test.lua
+++ b/test/box-luatest/gh_7988_auth_type_test.lua
@@ -69,10 +69,7 @@ g.before_test('test_replication', function(cg)
     cg.replica = server:new({
         alias = 'replica',
         box_cfg = {
-            replication = {
-                server.build_listen_uri('master'),
-                server.build_listen_uri('replica'),
-            },
+            replication = server.build_listen_uri('master'),
         },
     })
     cg.replica:start()
diff --git a/test/engine-luatest/gh_6436_field_constraint_test.lua b/test/engine-luatest/gh_6436_field_constraint_test.lua
index b6a46d171..1202d9b67 100644
--- a/test/engine-luatest/gh_6436_field_constraint_test.lua
+++ b/test/engine-luatest/gh_6436_field_constraint_test.lua
@@ -588,7 +588,7 @@ g.test_constraint_replication = function(cg)
 
     local replica_cfg = {
         replication = server.build_listen_uri('master'),
-        bootstrap_strategy = 'legacy',
+        replication_sync_timeout = 300,
     }
     local replica = server:new({alias = 'replica', box_cfg = replica_cfg})
     replica:start()
diff --git a/test/replication-luatest/gh_5158_invalid_recovery_from_xlog_test.lua b/test/replication-luatest/gh_5158_invalid_recovery_from_xlog_test.lua
index 2d6e7b925..9828a00a3 100644
--- a/test/replication-luatest/gh_5158_invalid_recovery_from_xlog_test.lua
+++ b/test/replication-luatest/gh_5158_invalid_recovery_from_xlog_test.lua
@@ -19,7 +19,6 @@ g.before_each(function(cg)
             replication = {
                 server.build_listen_uri('master'),
             },
-            bootstrap_strategy = 'legacy',
         },
     }
     cg.cluster:start()
diff --git a/test/replication-luatest/gh_7584_missing_xlog_test.lua b/test/replication-luatest/gh_7584_missing_xlog_test.lua
index d75c13eed..cc2adeaa3 100644
--- a/test/replication-luatest/gh_7584_missing_xlog_test.lua
+++ b/test/replication-luatest/gh_7584_missing_xlog_test.lua
@@ -17,7 +17,6 @@ g.before_all(function(cg)
         box_cfg = {
             replication = {
                 server.build_listen_uri('master'),
-                server.build_listen_uri('replica'),
             },
         },
     }
diff --git a/test/replication-luatest/iproto_id_error_test.lua b/test/replication-luatest/iproto_id_error_test.lua
index d37c8f5d7..dd85d526e 100644
--- a/test/replication-luatest/iproto_id_error_test.lua
+++ b/test/replication-luatest/iproto_id_error_test.lua
@@ -30,7 +30,9 @@ g.test_iproto_id_error = function(cg)
     cg.replica:exec(function(uri)
         local t = require('luatest')
         t.assert_is_not(box.info.replication[1].upstream, nil)
-        t.assert_equals(box.info.replication[1].upstream.status, 'follow')
+        t.helpers.retrying({}, function()
+            t.assert_equals(box.info.replication[1].upstream.status, 'follow')
+        end)
         box.cfg({replication = {}})
         t.assert_is(box.info.replication[1].upstream, nil)
         box.cfg({replication = uri})
diff --git a/test/replication-luatest/linearizable_test.lua b/test/replication-luatest/linearizable_test.lua
index 0ff7ea87c..64679a29e 100644
--- a/test/replication-luatest/linearizable_test.lua
+++ b/test/replication-luatest/linearizable_test.lua
@@ -185,7 +185,7 @@ g.test_reconnect_while_waiting = function(cg)
             box.cfg{replication = repl}
         end, {cg.box_cfg.replication})
     end
-    cluster:wait_for_fullmesh()
+    cg.cluster:wait_for_fullmesh()
     local success, ok, _ = f:join()
     t.assert(success, 'fiber joined successfully')
     t.assert(ok, 'reconnect while begin() waits for vclocks is tolerated')
diff --git a/test/replication-py/failed.lua b/test/replication-py/failed.lua
index 68ef10ec5..ed0ba38cc 100644
--- a/test/replication-py/failed.lua
+++ b/test/replication-py/failed.lua
@@ -4,7 +4,6 @@ box.cfg({
     listen              = os.getenv("LISTEN"),
     replication         = os.getenv("MASTER"),
     memtx_memory        = 107374182,
-    bootstrap_strategy  = "legacy",
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/replication-py/replica.lua b/test/replication-py/replica.lua
index 6d1f2e7f6..2c0554dfd 100644
--- a/test/replication-py/replica.lua
+++ b/test/replication-py/replica.lua
@@ -6,5 +6,5 @@ box.cfg({
     replication         = os.getenv("MASTER"),
     memtx_memory        = 107374182,
     replication_timeout = 0.1,
-    bootstrap_strategy  = "legacy",
+    bootstrap_strategy  = 'legacy',
 })
diff --git a/test/replication-py/uuid_mismatch.lua b/test/replication-py/uuid_mismatch.lua
index b366d809d..1d26bcda4 100644
--- a/test/replication-py/uuid_mismatch.lua
+++ b/test/replication-py/uuid_mismatch.lua
@@ -4,7 +4,6 @@ box.cfg({
     listen              = os.getenv("LISTEN"),
     replication         = os.getenv("MASTER"),
     replicaset_uuid     = "12345678-abcd-abcd-abcd-123456789000",
-    bootstrap_strategy  = "legacy",
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/replication/anon.lua b/test/replication/anon.lua
index 52a7f4e3e..2e7ee9983 100644
--- a/test/replication/anon.lua
+++ b/test/replication/anon.lua
@@ -8,7 +8,6 @@ box.cfg({
     replication_connect_timeout = 0.5,
     read_only=true,
     replication_anon=true,
-    bootstrap_strategy = 'legacy',
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/replication/gh-5213-replica.lua b/test/replication/gh-5213-replica.lua
index 10fe9bead..3ba90d7ab 100644
--- a/test/replication/gh-5213-replica.lua
+++ b/test/replication/gh-5213-replica.lua
@@ -5,7 +5,6 @@ box.cfg({
     replication = os.getenv("MASTER"),
     read_only = true,
     memtx_use_mvcc_engine = true,
-    bootstrap_strategy  = 'legacy',
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/replication/replica_auth.lua b/test/replication/replica_auth.lua
index c341b6ee7..61d046fc4 100644
--- a/test/replication/replica_auth.lua
+++ b/test/replication/replica_auth.lua
@@ -9,5 +9,4 @@ box.cfg({
     listen = os.getenv("LISTEN"),
     replication = USER_PASS .. "@" .. os.getenv("MASTER"),
     replication_timeout = TIMEOUT,
-    bootstrap_strategy  = 'legacy',
 })
diff --git a/test/replication/replica_on_schema_init.lua b/test/replication/replica_on_schema_init.lua
index e06f2e57c..eb5a5c992 100644
--- a/test/replication/replica_on_schema_init.lua
+++ b/test/replication/replica_on_schema_init.lua
@@ -20,7 +20,6 @@ end)
 box.cfg({
     listen              = os.getenv("LISTEN"),
     replication         = os.getenv("MASTER"),
-    bootstrap_strategy  = 'legacy',
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/replication/wal_off.lua b/test/replication/wal_off.lua
index 6db65cc2a..9da53ba98 100644
--- a/test/replication/wal_off.lua
+++ b/test/replication/wal_off.lua
@@ -6,7 +6,6 @@ box.cfg({
     memtx_memory        = 107374182,
     wal_mode            = 'none',
     replication_timeout = 0.1,
-    bootstrap_strategy  = 'legacy',
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/vinyl-luatest/gh_6568_replica_initial_join_removal_of_compacted_run_files_test.lua b/test/vinyl-luatest/gh_6568_replica_initial_join_removal_of_compacted_run_files_test.lua
index 131d5ae4b..887f80b14 100644
--- a/test/vinyl-luatest/gh_6568_replica_initial_join_removal_of_compacted_run_files_test.lua
+++ b/test/vinyl-luatest/gh_6568_replica_initial_join_removal_of_compacted_run_files_test.lua
@@ -13,7 +13,6 @@ g.before_all(function()
         vinyl_memory = 128,
         replication  = { g.master.net_box_uri },
         read_only    = true,
-        bootstrap_strategy = 'legacy',
     }
     g.replica = g.cluster:build_and_add_server({alias   = 'replica',
                                                 box_cfg = replica_box_cfg})
diff --git a/test/vinyl/replica_rejoin.lua b/test/vinyl/replica_rejoin.lua
index 42acf2dbd..f97209c44 100644
--- a/test/vinyl/replica_rejoin.lua
+++ b/test/vinyl/replica_rejoin.lua
@@ -9,7 +9,6 @@ box.cfg({
     replication         = replication,
     vinyl_memory        = 1024 * 1024,
     wal_cleanup_delay   = 0,
-    bootstrap_strategy  = 'legacy',
 })
 
 require('console').listen(os.getenv('ADMIN'))
diff --git a/test/xlog/replica.lua b/test/xlog/replica.lua
index eb2b4e4bf..c89eb989e 100644
--- a/test/xlog/replica.lua
+++ b/test/xlog/replica.lua
@@ -8,5 +8,4 @@ box.cfg({
     listen              = os.getenv("LISTEN"),
     replication         = os.getenv("MASTER"),
     memtx_memory        = 107374182,
-    bootstrap_strategy  = 'legacy'
 })

```